File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven minutesInOven = 
    expectedMinutesInOven - minutesInOven

let preparationTimeInMinutes numberOfLayers = 
    2 * numberOfLayers

let elapsedTimeInMinutes numberOfLayers minutesInOven = 
    (preparationTimeInMinutes numberOfLayers) + minutesInOven

File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    private const int MinutesInOven = 40;
    private const int MinutesToPrepareOneLayer = 2;

    public int ExpectedMinutesInOven() => MinutesInOven;

    public int RemainingMinutesInOven(int minutesInOven) =>
        ExpectedMinutesInOven() - minutesInOven;

    public int PreparationTimeInMinutes(int numberOfLayers) =>
        MinutesToPrepareOneLayer * numberOfLayers;

    public int ElapsedTimeInMinutes(int numberOfLayers, int minutesInOven) =>
        PreparationTimeInMinutes(numberOfLayers) + minutesInOven;
}
